sklearn/metrics/tests/test_ranking.py

```diff
@@ -662,14 +662,31 @@ def test_auc_score_non_binary_class():
             roc_auc_score(y_true, y_pred)
 
 
-def test_binary_clf_curve():
+def test_binary_clf_curve_multiclass_error():
     rng = check_random_state(404)
     y_true = rng.randint(0, 3, size=10)
     y_pred = rng.rand(10)
     msg = "multiclass format is not supported"
+
+    with pytest.raises(ValueError, match=msg):
+        precision_recall_curve(y_true, y_pred)
+
+    with pytest.raises(ValueError, match=msg):
+        roc_curve(y_true, y_pred)
+
+
+def test_binary_clf_curve_implicit_pos_label():
+    y_true = ["a", "b"]
+    y_pred = [0., 1.]
+    msg = ("make y_true take integer value in {0, 1} or {-1, 1}"
```

Comment:
We can check the `y_true takes value in ['a' 'b']` part of the message here.

(Or if we keep `classes` a set, then `y_true takes value in {'a' 'b'}` )

Reply:
I did not want to do this because the ordering of set elements is not deterministic in Python 3.5 which we still support. Maybe I can make an assertion conditional on the Python version.

Reply:
Actually set are always unordered. I will find another way to format this.